akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineProviderSpec.scala

```diff
@@ -261,14 +264,17 @@ abstract class RotatingKeysSSLEngineProviderSpec(extraConfig: String)
     (remoteSysB, pathEchoB)
   }
 
-  override def afterTermination(): Unit = {
+  override def beforeTermination(): Unit = {
     systemsToTerminate.foreach { systemToTerminate =>
       system.log.info(s"Terminating $systemToTerminate...")
-      Await.result(systemToTerminate.terminate(), 10.seconds)
+      val now = System.nanoTime()
+      Await.result(systemToTerminate.terminate(), 8.seconds.dilated)
+      val lasted = System.nanoTime() - now
+      system.log.info(s"Terminated $systemToTerminate after ${lasted / 1000000} ms")
     }
     // Don't cleanup folder until all systems have terminated
     cleanupTemporaryDirectory()
```

Comment:
I like the idea of terminating the `systemsToTerminate` in `beforeTermination` instead of `afterTermination` but `cleanupTemporaryDirectory` must happen in `afterTermination` always: the system in testkit also uses the rotating keys implementation so it could poll the file system and fail when the folders have been deleted.

For the rest it LGTM.